src/Doctrine/Orm/Extension/ParameterExtension.php

```diff
@@ -49,9 +54,20 @@ private function applyFilter(QueryBuilder $queryBuilder, QueryNameGeneratorInter
                 continue;
             }
 
-            $filter = $this->filterLocator->has($filterId) ? $this->filterLocator->get($filterId) : null;
+            if (\is_string($filterId) && $this->filterLocator->has($filterId)) {
+                $filter = $this->filterLocator->has($filterId) ? $this->filterLocator->get($filterId) : null;
+            }
+
+            if (\is_object($filterId)) {
+                $filter = $filterId;
+                $filter->setManagerRegistry($this->managerRegistry);
+                $filter->setProperties($values);
+            }
+
             if ($filter instanceof FilterInterface) {
-                $filter->apply($queryBuilder, $queryNameGenerator, $resourceClass, $operation, ['filters' => $values, 'parameter' => $parameter] + $context);
+                $filter->apply($queryBuilder, $queryNameGenerator, $resourceClass, $operation,
+                    array_merge(['filters' => $values, 'parameter' => $parameter], $context)
```

Comment:
why array_merge?

Reply:
In my opinion `array_merge` is more explicit but after some research, I will rollback